fees/beraborrow/index.ts

```diff
@@ -0,0 +1,233 @@
+    dailyFees.add(debtTokens[i], log.amount);
+  });
+
+// DenManager Redemption Fees
+
+const denManagers = tuples.flatMap(t => t.denManagers);
+
+const DenManagerRedemptionFees = await Promise.all(
+  denManagers.map(async (denManager) => {
+    const logs = await options.getLogs({
+      target: denManager,
+      eventAbi: "event RedemptionFee(address indexed _denManager, address indexed _redeemer, uint256 _attemptedDebtAmount, uint256 _actualDebtAmount, uint256 _collateralSent, uint256 _collateralFee)",
+      fromBlock: 10,
+    });
+
+    return logs.length > 0
+      ? logs.map((log: any) => log.collateralFee)
+      : [0];
+  })
+);
+
+dailyFees.add(assets, DenManagerRedemptionFees.flat());
+
+
+
+
+// Liquidation fee (Debt gas compensation - TEMPORARY)
+
+const logs = await options.getLogs({
+  target: NECT,
+  eventAbi: "event Transfer(address indexed from, address indexed to, uint256 value)",
+  topics: [
+    TRANSFER_TOPIC,
+    "0x000000000000000000000000" + NectGasPool.substring(2),
+    "0x000000000000000000000000" + FEE_RECEIVER.substring(2),
+  ],
+  fromBlock: 10
+})
+
+const total = logs.reduce((acc: BigNumber, log: any) => {
+    return acc.plus(new BigNumber(log[2]));
+  }, new BigNumber(0));
+
+dailyFees.add(NECT, total);
+
+// Liquidation fee (Collateral gas compensation - TEMPORARY)
+
+await Promise.all(
+    
+  tuples.map(async (tuple) => {
+    const denManagers = tuple.denManagers;
+    const collateral = tuple.collateral;
+
+    const logs = await options.getLogs({
+      target: collateral,
+      eventAbi: "event Transfer(address indexed from, address indexed to, uint256 value)",
+      topics: [
+        TRANSFER_TOPIC,
+        "0x000000000000000000000000" + denManagers[0].substring(2),
+        "0x000000000000000000000000" + FEE_RECEIVER.substring(2),
+      ],
+    });
+    
+    const total = logs.reduce((acc: BigNumber, log: any) => {
+      return acc.plus(new BigNumber(log[2]));
+    }, new BigNumber(0));
+
+    dailyFees.add(collateral, total);
+  })
+);
+
+  
+// PermissionlessPSM deposit fee (in NECT)
+
+await Promise.all(
+  PSMContracts.map(async (psm) => {
+    const logs = await options.getLogs({
+      target: psm,
+      eventAbi: "event Deposit(address indexed caller, address indexed stable, uint stableAmount, uint mintedNect, uint fee)",
+    });
+
+    const total = logs.reduce((acc, log) => acc.plus(new BigNumber(log[4])), new BigNumber(0));
+    if (total.isGreaterThan(0)) dailyFees.add(NECT, total);
+  })
+);
+
+
+
+// PermissionlessPSM withdrawal fee (fee is in indexed stable)
+// OLD AND new PSM
+
+await Promise.all(
+  PSMContracts.map(async (psm) => {
+    const logs = await options.getLogs({
+      target: psm,
+      eventAbi: "event Withdraw(address indexed caller, address indexed stable, uint stableAmount, uint burnedNect, uint fee)",
+      fromBlock: 1470110,
+    });
+
+    logs.forEach((log) => {
+      const stable = log[1];
```

Comment:
you can do, log.stable and log.fee here, and the check if the field exists is not needed?